### PR TITLE
Add list_models function to return a list of OTX available models & 'otx find' CLI command

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

- Add `list_models()` for showing available OTX models.
This allows API users to use the model provided by auto-configuration by selecting a different recipe provided by OTX.
- Add `otx find` to use list_models and print Recipe tables.
This allows CLI users to get information about the location of available configuration files provided by OTX.

### Use-Case
1. list_models() & Training with Engine API
image

2. `otx find` CLI commands
image
*Refer: Training from model name in CLI is not working for now.

### How to test

`python -m pytest tests/unit/engine/utils/test_api.py`
`otx find`

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/engine/utils/api.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import fnmatch
+import textwrap
+from pathlib import Path
+
+from otx.core.utils.imports import get_otx_root_path
+
+RECIPE_PATH = get_otx_root_path() / "recipe"
+
+
+def list_models(task: str | None = None, pattern: str | None = None, print_table: bool = False) -> list[str]:
```
Comment (vinnamkim) [inline src/otx/engine/utils/api.py]:
```suggestion
def list_models(task: OTXTaskType | None = None, pattern: str | None = None, print_table: bool = False) -> list[str]:
```

File: src/otx/engine/utils/api.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        # Return all available model list that matches the pattern.
+        >>> models = list_models(task="MULTI_CLASS_CLS", pattern="*efficient")
+        >>> models
+        ['otx_efficientnet_b0', 'efficientnet_v2_light', 'efficientnet_b0_light', ...]
+
+        # Print the recipe information as a Rich.Table (include task, model name, recipe path)
+        >>> models = list_models(task="MULTI_CLASS_CLS", pattern="*efficient", print_table=True)
+    """
+    task = task.lower() if task is not None else "**"
```
Comment (vinnamkim) [inline src/otx/engine/utils/api.py]:
```suggestion
    task = task.name.lower() if task is not None else "**"
```

File: src/otx/engine/utils/api.py (hunk 1 above)
Comment (harimkang) [reply src/otx/engine/utils/api.py]: Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/2856/commits/e22a3ace9f57830aaf1a5c24235f1f0a12ef78b6

File: src/otx/engine/utils/api.py (hunk 2 above)
Comment (harimkang) [reply src/otx/engine/utils/api.py]: Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/2856/commits/e22a3ace9f57830aaf1a5c24235f1f0a12ef78b6

**Review by vinnamkim — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**